### ISSUE TITLE
Bug in gVim 8.0 windows x64: cannot delete file marks or numbered marks

### ISSUE THREAD

**Issue by xc1427** [vim/vim#1339]
I downloaded my gVim from the nightly built windows gvim repository. Here is the vim version information:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Oct 18 2016 22:02:07)
MS-Windows 64-bit GUI version with OLE support
Included patches: 1-45
Compiled by appveyor@APPVYR-WIN
Huge version with GUI.
```

I simply cannot delete file marks by doing for example `:delmarks A` nor for the numbered marks.
After the deletion when I do `:marks`, the deleted mark is not in the list. But when I re-entrer gVim, the deleted marks reappear. When I looked into the `_viminfo` file, the mark is never deleted. 

I tried disabing loading _gvimrc and _vimrc files but the deletion still does not work.
I also tried the latest version of x64 windows gVim (8.0.0134) and the one bundled with windows Git (vim 7.4.1721 x86-64 nogui). No luck.

I am a little bit surprised because this should be the core of core features in Vim. I wonder if I am doing wrong. Could someone test on their machine ?

**Comment by chrisbra** [comment]
After you use delmarks command, use :wviminfo so that the deleted mark will be recorded in the viminfo file. Please ask user questions on the appropriate forums.

**Comment by xc1427** [comment]
> use :wviminfo so that the deleted mark will be recorded in the viminfo file. 

Thank you for your reply. @chrisbra  I actually had tried `:wviminfo` command. But the deleted mark is still retained. You can try that.

You have inspired me to dig more into the vim doc. So finally I make myself understand how vim works on file marks. To delete the existing file marks, we have to use the command `:wviminfo!` (the exclamation mark version) to **forecefully overwrite** the existing viminfo file. 

- This is how I think vim works:
    + When vim is quitted, `:wviminfo` is **automatically** fired. It is of interest when we want to exchange infomation between two instances of vim **while vim is running**. But we do not need to fire it manually to persist information when quitting vim.
    + `:wviminfo` **merges** the information. As the deleted mark is still in the viminfo file after command `:delmarks`, it will be merged in viminfo, actually leaving itself not deleted.

- IMHO, the behavior should be discussed @brammool @vim-ml :
    + I have passed hours to test before I fire a bug report. I do not want to pollute the forums either. I cannot say this is a bug. However, IMHO. I insist to reopen this issue. 
    + The `:delmarks` command should be one shot. the CRUD operation should be consistent. When a user creates or modifies a file mark, he does not do `:wviminfo[!]` to persist. This should be the same when he wants to delete.
    + When a merge is done, the deletion should be prioritized over existence.
    + Or at least, this should be mentioned in the documentation.

**Comment by vim-ml** [comment]
On 2016-12-28, xc1427 wrote:

>       □ When a merge is done, the deletion should be prioritized over
>         existence.

Why?  How do you know, or rather, how is Vim supposed to know, that
you prefer deletion over existence?

Say you have a .viminfo file that contains marks a and b.  Now you
open two instances of Vim.  Each reads that .viminfo file and stores
those two marks.  Now in the first Vim instance you create mark
c and close Vim.  Mark c is now added to .viminfo.  Now you close
the second Vim instance.  It sees mark c in the .viminfo file but
not in its memory.  Should it delete mark c?  I don't think so.

The point is that I'm sure you can find a scenario in which Vim's
behavior is not what you want and not what you think it should be,
but that doesn't mean that that behavior is wrong in all scenarios.

Regards,
Gary

**Comment by xc1427** [comment]
> Say you have a .viminfo file that contains marks a and b.  Now you
> open two instances of Vim.  Each reads that .viminfo file and stores
> those two marks.  Now in the first Vim instance you create mark
> c and close Vim.  Mark c is now added to .viminfo.  Now you close
> the second Vim instance.  It sees mark c in the .viminfo file but
> not in its memory.  Should it delete mark c?  I don't think so.

I agree with you that we do not delete mark c in this situation.

So I correct : **when a merge is done, the _latest operation_ should be taken.** I think that is one of the reasons why a **time-stamp** is added onto the mark in the recent vim release. This fits the deletion and your scenario.
- **Gary's Scenario**: Open two instances of vim. The first vim creates a mark _C_ and quit, then quits second one.
As the second vim has no time-stamp information  in memory on _C_ from itself, the latest operation is considered to be the creation. So no deletion. It's good.

- **Deletion Scenario**: Suppose we have already a mark _E_. Open one instances of vim. Do `delmarks:E` then quit.
The latest operation on _E_ is the deletion which has been fired by the only vim instance. Then _E_ is deleted, as I suggested.

In conclusion, to take the latest operation into account , IMHO, is quite a sensible and common idea where no lock mechanism is applicable. 

Regards,

**Comment by xc1427** [comment]
I am sorry that I did not see this post #448 and [this discussion](https://groups.google.com/forum/#!msg/vim_dev/kre4oyzMRDU/egIH4hlIWo4J) in google vim_dev.

The vim 8.0 new feature list says:
> Viminfo uses timestamps ~
> 
> Previously the information stored in viminfo was whatever the last Vim wrote
> there.  Now timestamps are used to always keep the most recent items.
> See |viminfo-timestamp|.

So we have time-stamp now, I think there is really no reason to keep the old behavior @brammool . As a user, he fires one deletion command and he expects the thing to disappear. It is that simple.

**Comment by chrisbra** [comment]
Actually I thought, that since we have timestamps, it would work with a simple :wviminfo.
